pkg/grpc/exec/exec_test_helper.go

```diff
@@ -415,7 +451,7 @@ func GrpcExecOutOfOrder[EXEC notify.Message, EXIT notify.Message](t *testing.T)
 		AllEvents = append(AllEvents, e)
 	}
 
-	time.Sleep(time.Millisecond * time.Duration((option.Config.EventCacheNumRetries+4)*CacheTimerMs)) // wait for cache to do it's work
+	dn.WaitNotifier(4)
```

Comment:
Small question: where does this argument (4) comes from? In other cases it seems to be 1 or 2. Why don't we have the same number of retries in all cases?

Reply:
it's the number of events we expect from the notifier